drools/src/main/java/com/datatorrent/drools/rules/DroolListener/RulesFiredListener.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2017 DataTorrent, Inc. ALL Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.drools.rules.DroolListener;
+
+import java.util.List;
+import java.util.Map;
+
+import org.kie.api.definition.rule.Rule;
+import org.kie.api.event.rule.AfterMatchFiredEvent;
+import org.kie.api.event.rule.AgendaEventListener;
+import org.kie.api.event.rule.AgendaGroupPoppedEvent;
+import org.kie.api.event.rule.AgendaGroupPushedEvent;
+import org.kie.api.event.rule.BeforeMatchFiredEvent;
+import org.kie.api.event.rule.MatchCancelledEvent;
+import org.kie.api.event.rule.MatchCreatedEvent;
+import org.kie.api.event.rule.RuleFlowGroupActivatedEvent;
+import org.kie.api.event.rule.RuleFlowGroupDeactivatedEvent;
+
+public class RulesFiredListener implements AgendaEventListener{
+
+  Map<Object,List<Rule>> factRule;
+  Map<Rule,Integer> ruleCount;
+
+  public RulesFiredListener(Map<Object, List<Rule>> factRule, Map<Rule, Integer> ruleCount)
+  {
+    this.factRule = factRule;
+    this.ruleCount = ruleCount;
+  }
+
+  RulesFiredListener()
+  {
+
+  }
+  public Map<Rule, Integer> getRuleCount()
+  {
+    return ruleCount;
+  }
+
+  public void setRuleCount(Map<Rule, Integer> ruleCount)
+  {
+    this.ruleCount = ruleCount;
+  }
+
+  public Map<Object, List<Rule>> getFactRule()
+  {
+    return factRule;
+  }
+
+  public void setFactRule(Map<Object, List<Rule>> factRule)
+  {
+    this.factRule = factRule;
+  }
+
+  /**
+   * This function will be called automatically just before firing a Rule.
+   * Modifies the map factRule and ruleCount.
+   * Note that in this map the fact will be modified once the rule is fired
+   * and we don't need to manually modify it.
+   *
+   * @param event the event which triggered this call
+   */
+  public void beforeMatchFired(BeforeMatchFiredEvent event)
+  {
+    Rule matchedRule = event.getMatch().getRule();
+    for (Object o : event.getMatch().getObjects()) {
+      List<Rule> tempList = factRule.get(o);
```

Comment:
please rename tempList to some meaningful name. Same for other variables like "o" and others in these code changes.